base/subarray.jl

```diff
@@ -433,6 +433,18 @@ find_extended_inds(::ScalarIndex, I...) = (@inline; find_extended_inds(I...))
 find_extended_inds(i1, I...) = (@inline; (i1, find_extended_inds(I...)...))
 find_extended_inds() = ()
 
+@inline function iterate(v::FastSubArray)
```

Comment:
are these `@inline` necessary? I'd be very surprised if they weren't automatically inlined.

Reply:
I tried it without `@inline`, and I didn't see a difference.

Reply:
`@inline` was added to avoid possible regression. (The default fallback has it.)
I have no specific example. But just imagine `v`'s parent has expensive `getindex` (e.g. 1d BroadcastArray).

Reply:
my guess is the fallback has it because it's from a time when we trusted our inlining algorithm a lot less.

Reply:
comment above said
```julia
# While the definitions for IndexLinear are all simple enough to inline on their
# own, IndexCartesian's CartesianIndices is more complicated and requires explicit
# inlining.
```
Of course there's no need to worry about `IndexCartesian`.
But I think explicit `@inline` is not a bad idea.
The recomputed `stride1` would be propagated only if `iterate` get inlined (IIRC the const info in `Core.PartialStruct` only works if the corresponding `getfield` happens at the same scope?)